File: src/class_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

class NumberCounter {

  private:
    int counter;
    ros::Publisher pub;
  
  public:
    NumberCounter(ros::NodeHandle *nh) {
      counter = 0;
      pub = nh->advertise<std_msgs::Int64>("/counter", 1);    
    }
  
    void pub_function() {
      counter = counter + 1;
      std_msgs::Int64 msg;
      msg.data = counter;
      pub.publish(msg);
    }
};

int main (int argc, char **argv)
{
  ros::init(argc, argv, "pub_counter");
  ros::NodeHandle nh;
  NumberCounter nc = NumberCounter(&nh);
  ros::WallRate r(1); // 1 Hz
  while (ros::ok()) {
    nc.pub_function();
    ros::spinOnce();
    r.sleep();
  }
}

File: src/class_publisher.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64


class NumberCounter():

  def __init__(self):
    self.count = 0
    self.pub = rospy.Publisher("counter", Float64, queue_size=1)


if __name__ == "__main__":

  rospy.init_node('publisher')
  pub_class = NumberCounter()
  rate = rospy.Rate(2)  # 2 Hz

  pub_class.count = 0

  while not rospy.is_shutdown():

    result = Float64()
    result.data = pub_class.count
    rospy.loginfo(str(result.data))
    pub_class.pub.publish(result)
    pub_class.count = pub_class.count + 1
    rate.sleep()
